cli/azd/pkg/auth/manager_test.go

```diff
@@ -0,0 +1,82 @@
+package auth
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadUserProperties(t *testing.T) {
+	t.Run("homeID", func(t *testing.T) {
+		cfg := config.NewConfig(nil)
+		require.NoError(t, cfg.Set("auth.account.currentUser.homeAccountId", "testAccountId"))
+
+		props, err := readUserProperties(cfg)
+
+		require.NoError(t, err)
+		require.Nil(t, props.ClientID)
+		require.Nil(t, props.TenantID)
+		require.Equal(t, "testAccountId", *props.HomeAccountID)
+	})
+
+	t.Run("clientID", func(t *testing.T) {
+		cfg := config.NewConfig(nil)
+		require.NoError(t, cfg.Set("auth.account.currentUser.clientId", "testClientId"))
+		require.NoError(t, cfg.Set("auth.account.currentUser.tenantId", "testTenantId"))
+
+		props, err := readUserProperties(cfg)
+
+		require.NoError(t, err)
+		require.Nil(t, props.HomeAccountID)
+		require.Equal(t, "testClientId", *props.ClientID)
+		require.Equal(t, "testTenantId", *props.TenantID)
+	})
+}
+
+func TestServicePrincipalLoginClientSecret(t *testing.T) {
+	credentialCache := &memoryCache{
+		cache: make(map[string][]byte),
+	}
+
+	m := Manager{
+		configManager:   &memoryConfigManager{},
+		credentialCache: credentialCache,
+	}
+
+	cred, err := m.LoginWithServicePrincipalSecret(
+		context.Background(), "testClientId", "testTenantId", "testClientSecret",
+	)
+
+	require.NoError(t, err)
+	require.IsType(t, cred, new(azidentity.ClientSecretCredential))
+
+	cred, err = m.CredentialForCurrentUser(context.Background())
+
+	require.NoError(t, err)
+	require.IsType(t, cred, new(azidentity.ClientSecretCredential))
+}
+
+type memoryConfigManager struct {
+	configs map[string]config.Config
+}
+
+func (m *memoryConfigManager) Load(filePath string) (config.Config, error) {
```

Comment:
FYI - there is a `MockConfigManager` that I'm using in some other tests.  It basically just takes  a config value to return.  Take a look to see if you are able to leverage this or if we need to implement this new version.

Reply:
I think I need something a little more durable than what MockConfigManager provides.  I need `Save` to be implemented in some fashion for some of these tests (it would be fine if Save on MockConfigManager was just "overwrite the object I hand back for every call to Load).